qiskit_experiments/experiment_data.py

```diff
@@ -12,130 +12,321 @@
         """
-        # Experiment identification metadata
-        self._id = str(uuid.uuid4())
+        # Experiment class object
         self._experiment = experiment
 
-        # Experiment Data
+        # Terra ExperimentDataV1 attributes
+        self._backend = backend
+        self._id = str(uuid.uuid4())
+        if experiment is not None:
+            self._type = experiment._type
+        else:
+            self._type = None
+        job_ids = job_ids or []
+        self._jobs = OrderedDict((k, None) for k in job_ids)
         self._data = []
-
-        # Analysis
+        self._figures = OrderedDict()
+        self._figure_names = []
         self._analysis_results = []
 
-    def __repr__(self):
-        line = 51 * "-"
-        n_res = len(self._analysis_results)
-        ret = line
-        ret += f"\nExperiment: {self._experiment._type}"
-        ret += f"\nExperiment ID: {self.experiment_id}"
-        ret += "\nStatus: COMPLETE"
-        ret += f"\nCircuits: {len(self._data)}"
-        ret += f"\nAnalysis Results: {n_res}"
-        ret += "\n" + line
-        if n_res:
-            ret += "\nLast Analysis Result"
-            for key, value in self._analysis_results[-1].items():
-                ret += f"\n- {key}: {value}"
-        return ret
-
     @property
-    def experiment_id(self):
-        """Return the experiment id"""
-        return self._id
-
-    def experiment(self):
+    def experiment(self) -> "BaseExperiment":
         """Return Experiment object"""
         return self._experiment
 
-    def analysis_result(self, index):
-        """Return stored analysis results
-
-        Args:
-            index (int or slice): the result or range of results to return.
-
-        Returns:
-            AnalysisResult: the result for an integer index.
-            List[AnalysisResult]: a list of results for slice index.
-        """
-        return self._analysis_results[index]
+    @property
+    def experiment_type(self) -> str:
+        """Return the experiment type"""
+        return self._type
 
-    def add_analysis_result(self, result):
-        """Add an Analysis Result
+    @property
+    def experiment_id(self) -> str:
+        """Return the experiment id"""
+        return self._id
 
-        Args:
-            result (AnalysisResult): the analysis result to add.
+    @property
+    def job_ids(self) -> List[str]:
+        """Return experiment job IDs.
+        Returns: IDs of jobs submitted for this experiment.
         """
-        self._analysis_results.append(result)
+        return list(self._jobs.keys())
 
     @property
-    def data(self):
-        """Return stored experiment data"""
-        return self._data
-
-    def add_data(self, data):
-        """Add data to the experiment.
+    def backend(self) -> Union["BaseBackend", "Backend"]:
+        """Return backend.
+        Returns:
+            Backend this experiment is for.
+        """
+        return self._backend
 
+    def add_data(
```

Comment:
I see that you copied a lot of sections from the Terra PR. I assume that this is a temporary copy, and that once the Terra PR is merged and we inherit from the class there, these sections will be removed.

Reply:
Yep exactly

Reply:
I have a follow up branch to this PR where I started to do this, but its needs a few small changes to the Terra PR that I left as review there: https://github.com/chriseclectic/qiskit-experiments/blob/resultdb/qiskit_experiments/experiment_data.py#L34

Reply:
Please be aware of #29